pkg/upgrade/base.go

```diff
@@ -13,24 +14,27 @@ import (
 	kubernetes2 "github.com/vmware/cluster-api-upgrade-tool/pkg/internal/kubernetes"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	clusterapiv1alpha1client "sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
-	"sigs.k8s.io/cluster-api/pkg/controller/noderefutil"
+	bootstrapv1 "sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/cluster-api/controllers/noderefutil"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type base struct {
-	log                        logr.Logger
-	userVersion                semver.Version
-	desiredVersion             semver.Version
-	clusterNamespace           string
-	clusterName                string
-	managementClusterAPIClient clusterapiv1alpha1client.ClusterV1alpha1Interface
-	targetRestConfig           *rest.Config
-	targetKubernetesClient     kubernetes.Interface
-	providerIDsToNodes         map[string]*v1.Node
-	imageField, imageID        string
-	upgradeID                  string
+	log                    logr.Logger
+	userVersion            semver.Version
+	desiredVersion         semver.Version
+	clusterNamespace       string
+	clusterName            string
+	managerClusterClient   ctrlclient.Client
```

Comment:
Same here, this could be just `client`?

Reply:
I think it makes sense to assert that its the manager client